### ISSUE TITLE
Hi, I have a question, what the num_class and  num_classes mean?  

### ISSUE THREAD

**Issue by shuxiao9058** [eric612/MobileNet-YOLO#12]
As titled, for example, I want to  train only 1 type of object, how can I set the numbers in train and test protoxt files? 2 or 1?

**Comment by eric612** [comment]
You can set classes = 1 , and prev convolution channel = (5+1)*3  , if loss function is yolov3

**Comment by shuxiao9058** [comment]
I'm a beginner with DL, now I work with yolo lite prototxt you provide.
Thanks for your reply.

**Comment by shuxiao9058** [comment]
How can I set the prev convolution channel? @eric612

**Comment by eric612** [comment]
You can search the text in test and train prototxt and replace
num_output: 75 to num_output: 18

**Comment by shuxiao9058** [comment]
Thanks, I will have a try. How about the num_output with yolov2?

I have search on the net, and got:
[](https://github.com/pjreddie/darknet/issues/561)
[](https://github.com/pjreddie/darknet/issues/323)
[](https://www.bbsmax.com/A/VGzlVjQy5b/)

So if trained with yolov2:
coords=4
num_class=1
(classes+ coords+ 1)* (NUM)=(1+ 4+ 1)*5=30

Is it right? and the NUM is in num of RegionLoss layer.

**Comment by shuxiao9058** [comment]
But it is crashed with this error:

```
F0905 05:51:21.390486 22399 base_conv_layer.cpp:121] Check failed: channels_ % group_ == 0 (18 vs. 0)
*** Check failure stack trace: ***
    @     0x7fc86a3edbfd  google::LogMessage::Fail()
    @     0x7fc86a3ef9cc  google::LogMessage::SendToLog()
    @     0x7fc86a3ed7ec  google::LogMessage::Flush()
    @     0x7fc86a3f02de  google::LogMessageFatal::~LogMessageFatal()
    @     0x7fc86ae701fa  caffe::BaseConvolutionLayer<>::LayerSetUp()
    @     0x7fc86afe2d05  caffe::Net<>::Init()
    @     0x7fc86afe4be2  caffe::Net<>::Net()
    @     0x7fc86aff2f25  caffe::Solver<>::InitTrainNet()
    @     0x7fc86aff4083  caffe::Solver<>::Init()
    @     0x7fc86aff435f  caffe::Solver<>::Solver()
    @     0x7fc86afcd6e4  caffe::Creator_RMSPropSolver<>()
    @           0x411a3d  caffe::SolverRegistry<>::CreateSolver()
    @           0x40abcc  train()
    @           0x4085fc  main
    @     0x7fc85246fc05  __libc_start_main
    @           0x408e6b  (unknown)
```

**Comment by eric612** [comment]
It means there has no ground truth box match one of the yolo loss , so it will return null
